plugins/UpdateChecker/UpdateChecker.py

```diff
@@ -32,7 +32,7 @@ class UpdateChecker(Extension):
     The plugin is currently only usable for applications maintained by Ultimaker. But it should be relatively easy
     to change it to work for other applications.
     """
-    url = "https://software.ultimaker.com/latest.json"
+    url = "https://raw.githubusercontent.com/Ultimaker/Uranium/CURA-9272_semver_postfix/tests/latest.json"
```

Comment:
Why do we even have this hard coded? and why should Uranium be aware of Cura being able to update?
I would suggest to add this link to the https://github.com/Ultimaker/Cura/blob/main/cura/UltimakerCloud/UltimakerCloudConstants.py . You can also add it as an option to the conanfile.py of Cura. This would allow us to pass the endpoint dynamically, the same way we switch staging endpoints.

Make sure that the logic in this repo doesn't fail when an empty/default url exist here.

Reply:
I honestly have no idea, maybe this whole file should be moved into Cura? Otherwise, it will not be possible to fetch any configuration information without coupling Uranium to Cura.